### PR TITLE
Define BMD gene list within GWAS SNP LD blocks

### PR DESCRIPTION
**Pull request by gwaybio**
Here, I add functionality to assign genes to LD blocks of association (genomic ranges of r2 > 0.4 for bone mineral density GWAS SNPs)

### CODE REVIEW

File: scripts/generate_index_files.py [hunk 1]
```diff
@@ -84,11 +86,15 @@ def curate_tad_elements(tad_df, input_df, gen_class):
         if gen_class == 'snp' and chrom != 'X':
             chrom = int(chrom)
 
+        if gen_class == 'LD':
+                    chrom = 'chr{}'.format(chrom)
```
Comment (dhimmel) [inline scripts/generate_index_files.py]: Overindentation

File: scripts/generate_index_files.py [hunk 2]
```diff
@@ -105,10 +111,16 @@ def curate_tad_elements(tad_df, input_df, gen_class):
                                     ignore_index=True)
             bnd_df = bnd_df.append(o_df, ignore_index=True)
 
-        # Assign TAD information
-        elem_sub_df['TAD_id'] = tad_id
-        elem_sub_df['TAD_start'] = start
-        elem_sub_df['TAD_end'] = end
+        if gen_class == 'LD':
```
Comment (dhimmel) [inline scripts/generate_index_files.py]: These scripts should be split into modules with the actual calls being doone in notebooks as evidenced by this special casing and the hardcoding above:

```python
 if TAD_CELL in ['hESC', 'IMR90']:
     BASE_DIR = 'data/hg/'
     GENOME = 'hg19'
     REF_GENE_GTF = BASE_DIR + 'gencode.v19.annotation.gtf.gz'
     TAD_LOC = BASE_DIR + TAD_CELL + '_domains_hg19.bed'
     REF_SNPS = BASE_DIR + 'hg_common-snps.tsv'
 elif TAD_CELL in ['mESC', 'cortex']:
     BASE_DIR = 'data/mm/'
     GENOME = 'mm9'
     REF_GENE_GTF = BASE_DIR + 'gencode.vM9.annotation.gtf.gz'
     TAD_LOC = BASE_DIR + TAD_CELL + '_domains_mm9.bed'
     REF_SNPS = BASE_DIR + 'mm_common-snps.tsv'
 else:
     raise ValueError('Please input either "hESC", "IMR90", "mESC", or '
```

Outside of scope of this PR. Something to keep in mind for the future.

Comment (gwaygenomics) [inline scripts/generate_index_files.py]: you're saying here that each TAD (4 in total) should have its own notebook?

File: scripts/generate_index_files.py (hunk 1 above)
Comment (gwaygenomics) [inline scripts/generate_index_files.py]: thanks, nice catch

File: scripts/generate_index_files.py (hunk 2 above)
Comment (dhimmel) [inline scripts/generate_index_files.py]: Perhaps or that each TAD should just have a section in a single notebook. Just that the point of a script with arguments is to flexibly perform a certain analysis. This level of hardcoding makes me wonder why you accept arguments at all?

I think the best time to address these issues would be if you're creating a new repo and we brainstorm the analysis layout. Not really worth retrospectively modifying this.

**Review by dhimmel — APPROVED** [review]

File: scripts/generate_index_files.py (hunk 2 above)
Comment (gwaygenomics) [inline scripts/generate_index_files.py]: I agree, sounds good

**Pull request merged**